simbad/command_line/simbad_database.py

```diff
@@ -163,6 +189,151 @@ def create_lattice_db(database):
     np.savez_compressed(database, niggli_data)
 
 
+def create_contaminant_db(database, nproc=2, submit_qtype=None, submit_queue=False):
+    """Create a contaminant database
+
+    Parameters
+    ----------
+    database : str
+       The path to the database folder
+    nproc : int, optional
+       The number of processors [default: 2]
+    submit_qtype : str
+       The cluster submission queue type - currently support SGE and LSF
+    submit_queue : str
+       The queue to submit to on the cluster
+
+    Raises
+    ------
+    RuntimeError
+        dimple.contaminants.prepare module not available
+    RuntimeError
+       Windows is currently not supported
+    """
+
+    try:
+        import dimple.contaminants.prepare
+    except ImportError:
+        msg = "This feature will be available with the next dimple update"
+        raise RuntimeError(msg)
+
+    if CUSTOM_PLATFORM == "windows":
+        msg = "Windows is currently not supported"
+        raise RuntimeError(msg)
+
+    dimple.contaminants.prepare.main(verbose=False)
+
+    simbad_dat_path = os.path.join(database, '*', '*', '*', '*.dat')
+    existing_dat_files = [os.path.basename(f).split('.')[0] for f in glob.glob(simbad_dat_path)]
```

Comment:
We should look into the `glob2` module for this. It can do recursive searches. 

The code here would be changed to
```python
simbad_dat_path = os.path.join(database, '**', '*.dat')
existing_dat_files = [os.path.basename(f).split('.')[0] for f in glob2.iglob(simbad_dat_path)]
```

The `iglob` would create an iterator which is a little more efficient than storing the entire list in memory first.

Reply:
glob2 isn't in ccp4-python, I used the same code as in the morda database code so I expect you probably encountered the same issue when you were writing that.

Reply:
I know `glob2` isn't. Maybe Ronan could include it. We can definitely use `glob.iglob` though.

Reply:
I've updated this to use iglob but I've left `simbad_dat_path = os.path.join(database, '*', '*', '*', '*.dat')` as glob doesn't have a recursive feature